### ISSUE TITLE
allow templating events into native python types in deployment trigger parameters

### ISSUE THREAD

**Issue by zzstoatzz** [PrefectHQ/prefect#14875]
### Bug summary

right now, if i have something like
```yaml
- name: label-issues
  triggers:
    - enabled: true
      expect:
        - marvin.issue.opened
        - marvin.issue.reopened
      parameters:
        event_body_str: "{{ event.payload.body }}"
   ...
```

I would expect to be able to do
```python
@flow(log_prints=True)
async def label_issues(issue_event: GitHubIssueEvent)
```
and let the flow take me from json -> Model instance

but I need to do
```python
@flow(log_prints=True)
async def label_issues(event_body_str: str):
   issue_event = GitHubIssueEvent.model_validate_json(event_body_str)
```
because all the parameters get templated in as a `str`

### Version info (`prefect version` output)

```Text
» prefect version
Version:             3.0.0rc15+3.g4794589d89
API version:         0.8.4
Python version:      3.12.3
Git commit:          4794589d
Built:               Fri, Aug 9, 2024 2:40 PM
OS/Arch:             darwin/arm64
Profile:             pong
Server type:         cloud
Pydantic version:    2.8.2
Integrations:
  prefect-shell:     0.2.6
  prefect-dbt:       0.5.1
```

### Additional context

MRE
```python
from threading import Thread

from prefect import flow
from prefect.events import DeploymentEventTrigger, emit_event

trigger = DeploymentEventTrigger(  # type: ignore
    expect={"whatever.happened"}, parameters={"d": "{{ event.payload }}"}
)


@flow(log_prints=True)
def f(d: dict):
    print(d)


if __name__ == "__main__":
    try:
        t = Thread(target=f.serve, kwargs={"name": "test-flow", "triggers": [trigger]})
        t.start()
        while True:
            input("Press Enter to send event...\n\n")
            emit_event("whatever.happened", {"prefect.resource.id": str(id(__file__))})
    finally:
        t.join()
```

**Comment by kevingrismore** [comment]
I think having this would go a long way toward making the trigger->event->flow parameter story much cleaner. Having to modify a json string to convert it into a usable dictionary after it arrives feels just outside the nice experience I'd want to have using event data in my flows.

**Comment by zzstoatzz** [comment]
@kevingrismore big plus one on this

### notes to self
i was just looking at this, and it reminded me that we are checking the parameter schema server side, which means unfortunately we can't just be clever about in `validate_parameters`

```json
{
  "payload": {
    "action_index": 0,
    "action_type": "run-deployment",
    "invocation": "57cfd0a1-8b05-48c4-897d-90903636656b",
    "reason": "Validation error occurred for 'run-deployment' - Error creating flow run: Validation failed for field 'd'. Failure reason: '{}' is not of type 'object'",
    "validation_error": "Error creating flow run: Validation failed for field 'd'. Failure reason: '{}' is not of type 'object'",
    "status_code": 409
  },
}
  ```

**Comment by zzstoatzz** [comment]
closed by #15132
